work_queue/src/work_queue.c

```diff
@@ -3719,6 +3719,11 @@ static int check_hand_against_task(struct work_queue *q, struct work_queue_worke
 	if(w->resources->gpus.inuse + limits->gpus > overcommitted_resource_total(q, w->resources->gpus.total, 0)) {
 		ok = 0;
 	}
+		
+	//if the wall time for the worker is specified and there's not enough time for the task, then not ok
+	if(w->resources->time_left != -1 && w->resources->time_left < t->resources_requested->wall_time){
```

Comment:
Here we should also check for t->resources_requested->end (if > 0).  end is an absolute time, so something like:
t->resources->time_left > t->resources_requested->end - timestamp(). or something like that.

Also, not all tasks define -> wall_time, so you have to check for wall_time > 0
I would also change time_left != -1 to > 0 for consistency.